exercise/exercise_models.py

```diff
@@ -694,7 +696,9 @@ def grade(self, request, submission, no_penalties=False, url_name="exercise"):
         """
         Loads the exercise feedback page.
         """
-        language = get_language()
+        # Get the language from the submission
+        language = submission.meta_data.get('lang', None) or get_language()
```

Comment:
You could consider if the `submission` should expose the language via a property. Might be nice to use `submission.lang` here.

In addition, should we store the `get_language()` information in the database, thus making resubmissions consistent too? (did we have an issue about that?).

Reply:
That's interesting. Indeed, we could store the language in the meta data even if the original submission did not contain any language meta data. It would make sense for regrading submissions. I don't remember if we have had issues with that, but it makes sense that regrading can only use data from the submission. It should not take the language from the teacher's web session when the regrading is activated.

Reply:
For this PR, I could accept a TODO/FIXME comment here, marking that the `get_language` should be moved elsewhere (i.e. so that it's done when storing the submission to db and not when sending to grader).